### ISSUE TITLE
Duplicate cross section names

### ISSUE THREAD

**Issue by sclaw** [Dewberry/ripple1d#225]
HEC-RAS can only take cross section names of length 8 characters.  To fix cases when a sub model would have name greater than 8 length, names are stored as floats and rounded to the nearest whole number.  This can create instances of duplicate cross section names, especially when an engineer elected to put their names in units of miles.  RAS models with duplicate cross sections are invalid and will not run.  See example below.  Suggested: use autoincremented station names instead of preserving original stationing/names.

Collection: mip_02050107
Model: SusquehRvr_DanvilleLAMP
Sub model: 8153841

Source model:
![image](https://github.com/user-attachments/assets/43861600-d3d4-4f10-986d-3f3ab2b3612d)

Sub model (does not run):
![image](https://github.com/user-attachments/assets/39536e21-232b-4b1a-9332-098439cd4839)

**Comment by ar-siddiqui** [comment]
Did the original model have duplicate names? I must be missing something but why can't we just use what was in the original model?

**Comment by sclaw** [comment]
The original model did not have duplicate names.  When this sub model is created, it adjusts stationing from the most d/s cross section (581386).  The second cross-section was originally 0.16, but it becomes 581386.16.  Since 581386.16 is more than 8 characters (RAS name length limit), ripple1d rounds the name to 581386.0.

**Comment by ar-siddiqui** [comment]
Is there a reason why we are renaming cross-sections and not using original cross section names?

**Comment by sclaw** [comment]
I think that decision came out of a conversation before I joined.  @mdeshotel  can you weigh in?

**Comment by mdeshotel** [comment]
We have to ensure that river stationing increases as we move upstream (RAS requirement). So if the most downstream cross section on a tributary originally started at 1000 but the main stem (whose cross section we are also using) has a station of 500000 we need to increase the tributary river station to be 500000 + 1000 to ensure river stationing increases as we move upstream.

**Comment by ar-siddiqui** [comment]
Ah so RAS does not use station names in calculations but still wants them in order? weird.

**Comment by mdeshotel** [comment]
yes that is correct.  -very weird. furthermore, I found that you can sort of hack RAS by passing a non-numeric character to the river stationing (via the geometry text file) which it will accept.. but we would run into similar character limit issues if we tried to prepended some random non-numeric character to the source model stationing.

**Comment by ar-siddiqui** [comment]
I don't have any issues with the solution you recommended here. @sclaw do you take any issue with this solution?
We should document why we did what we did in technical documentation.

> As I mentioned previously, because of these complications, I would recommend that we rename the river stations to be 1,2,3,4,5, etc. Simple, clean, and less chance for these issues.
>

**Comment by sclaw** [comment]
It think that's a great solution.
